src/xml_parser_main_metadata_together.c

```diff
@@ -462,7 +462,11 @@ int cr_xml_parse_main_metadata_together(const char *primary_path,
     if (allow_out_of_order) {
         g_hash_table_destroy(cbdata.in_progress_pkgs_hash);
     } else {
-        cr_slist_free_full(cbdata.in_progress_pkgs_list, (GDestroyNotify) cr_package_free);
+        if (cbdata.newpkgcb) {
+            g_slist_free(cbdata.in_progress_pkgs_list);
+        } else {
+            cr_slist_free_full(cbdata.in_progress_pkgs_list, (GDestroyNotify) cr_package_free);
+        }
```

Comment:
IIRC - and it's been a while since I last looked at this - my struggle was that I'm not sure this would work for anyone using the C API directly, since they wouldn't have the Python layer to de-allocate the packages for them.

Reply:
Right, I think that is fine though.

The way I see it the user (this can be our Python bindings or any other C API direct user) is responsible for managing the packages if they created them.

I think it wouldn't be nice to free a package that was passed to `createrepo_c` in a callback, the user (even C API user) can still have a reference to it. In fact they should have it I think that is the main use case for `newpkgcb`.

I will add a note about this to the doc string.